src/main/java/com/conveyal/datatools/editor/controllers/api/EditorController.java

```diff
@@ -217,7 +221,7 @@ private String patchTable(Request req, Response res) {
         } catch (HaltException e) {
             throw e;
         } catch (Exception e) {
-            logMessageAndHalt(req, 500, "Could not patch update table", e);
+            logMessageAndHalt(req, 400, "Could not patch update table", e);
```

Comment:
I disagree, Evan. Did you see the original ticket (#257)? The UI throws up a very generic response when a 5xx error is encountered. This is OK because 500 errors should only be thrown when something very mysterious happened that's really outside the bounds of our application (e.g., networking level stuff). However, in this case, we have the exception in question, so we should pass it back to the requestor so that it gets displayed correctly. Generally, we can expect that the client request had something incorrect if this catch block is triggered.

Reply:
I did read the original ticket and am aware of what happens in terms of datatools application logic when a 5xx error is returned. However, as I noted, there are a lot of SQL Exceptions that could occur due to improper system problems that an end-user is unable to control. This could be something like the database user hasn't had permissions setup properly or the postgres database crashed and/or is currently unavailable. Those are unlikely, but still possible and therefore it'd be inappropriate to return a 4xx error because it wasn't the user's fault in those cases.